pkg/controller/signup.go

```diff
@@ -115,11 +115,22 @@ func (s *Signup) UpdateVerificationHandler(ctx *gin.Context) {
 	}
 
 	signup, err = s.verificationService.InitVerification(ctx, signup, phone.CountryCode, phone.PhoneNumber)
+	_, err2 := s.signupService.UpdateUserSignup(signup)
```

Comment:
I'm curious - why do we need to return the `UserSignup` instances from the methods when we pass the same `UserSignup` instances as a pointer

Reply:
It's not necessary, however I'm not going to change this here (see my other comment above).  I can address this in a separate PR, we are planning an architectural refactor to clean up the entire verification workflow.

Reply:
ok, I didn't know that you are planning an architectural refactor to clean up the code :+1: